lxd/device/nic.go

```diff
@@ -11,7 +11,7 @@ func nicValidationRules(requiredFields []string, optionalFields []string) map[st
 		"name":                    validate.IsAny,
 		"parent":                  validate.IsAny,
 		"network":                 validate.IsAny,
-		"mtu":                     validate.IsAny,
+		"mtu":                     validate.Optional(validate.IsUint32),
```

Comment:
@stgraber I wanted to raise this, as there appears to be some inconsistency in our validation of MTU.

For `bridge` network driver we validate it as signed int64 https://github.com/lxc/lxd/blob/master/lxd/network/driver_bridge.go#L162, however for `bridged` NICs (and various other places) we parse it as 32bit unsigned int:

https://github.com/lxc/lxd/blob/master/lxd/device/device_utils_network.go#L257
https://github.com/lxc/lxd/blob/master/lxd/device/device_utils_network.go#L313
https://github.com/lxc/lxd/blob/master/lxd/device/device_utils_network.go#L191

Admittedly those places do seem to all be written by myself, so there's a good chance I'm missing something here and MTU should be validated as something other than uint32.

But I'd like to make that consistent as possible.

Reply:
Negative MTU shouldn't be a thing, we should add a validate.IsMTU that validates >= 1280 && <= 9202.
Anything below 68 and the kernel doesn't allow IPv4, anything below 1280 and the kernel doesn't allow IPv6.

So let's avoid ourselves weird issues and require an IPv6-compatible MTU as the low value and cap at the max ethernet jumbo frame size.

Reply:
@stgraber great, will do and will apply to all NICs and bridge network settings.

Reply:
@stgraber oh the only other thing I should note is that we use the same NIC validation rules for infiniband devices, so would the rule you describe above apply to them as well?

Reply:
This https://community.mellanox.com/s/article/mtu-considerations-for-roce-based-applications suggests there are fixed MTU sizes for infiniband, and some of them are smaller than 1280.

Reply:
And https://wiki.archlinux.org/index.php/InfiniBand#MTU suggests one can use a much higher value of 65520

Reply:
@tomponline note that we only really care about IPoIB so the same minimum should apply, for the maximum, we could indeed go higher though again, that seems unlikely to matter in an IP world as you'll likely need to interact with non-IB IP devices at which point the 9202 cap will apply again.

So we can probably naively do 1280 - 9202 for everything now and if that becomes a problem, we can add some options.

Reply:
Great makes sense, thanks.